src/components/SDFlatButton.react.js

```diff
@@ -0,0 +1,222 @@
+   * that it passes all of its given props through to the underlying DOM
+   * element and renders its children prop for proper integration.
+   */
+  containerElement: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+
+  /**
+   * If true, the element's ripple effect will be disabled.
+   */
+  disableTouchRipple: PropTypes.bool,
+
+  /**
+   * Disables the button if set to true.
+   */
+  disabled: PropTypes.bool,
+
+  /**
+   * If true, the button will take up the full width of its container.
+   */
+  fullWidth: PropTypes.bool,
+
+  /**
+   * Color of button when mouse hovers over.
+   */
+  hoverColor: PropTypes.string,
+
+  /**
+   * The URL to link to when the button is clicked.
+   */
+  href: PropTypes.string,
+
+  /**
+   * Use this property to display an icon.
+   */
+  icon: PropTypes.node,
+
+  /* The element's ID */
+  id: PropTypes.string,
+
+  /**
+   * Label for the button.
+   */
+  label: PropTypes.string,
+
+  /**
+   * Place label before or after the passed children.
+   */
+  labelPosition: PropTypes.oneOf(['before', 'after']),
+
+  /**
+   * Override the inline-styles of the button's label element.
+   */
+  labelStyle: PropTypes.object,
+
+  /**
+   * An integer that represents the number of times
+   * that this element has been clicked on.
+   */
+  n_clicks: PropTypes.integer,
+
+  /**
+   * Callback function fired when the button is clicked.
+   *
+   * @param {object} event Click event targeting the button.
+   */
+
+  onClick: PropTypes.func,
+
+  /**
+   * Callback function fired when the element is focused or blurred by the keyboard.
+   *
+   * @param {object} event `focus` or `blur` event targeting the element.
+   * @param {boolean} isKeyboardFocused Indicates whether the element is focused.
+   */
+  onKeyboardFocus: PropTypes.func,
+
+  /**
+   * If true, colors button according to
+   * primaryTextColor from the Theme.
+   */
+  primary: PropTypes.bool,
+
+  /**
+   * Color for the ripple after button is clicked.
+   */
+  rippleColor: PropTypes.string,
+
+  /**
+   * If true, colors button according to secondaryTextColor from the theme.
+   * The primary prop has precendent if set to true.
+   */
+  secondary: PropTypes.bool,
+
+  /** Dash callback to update props on the server */
+  setProps: PropTypes.func,
+
+  /**
+   * Override the inline-styles of the root element.
+   */
+  style: PropTypes.object,
+};
+
+const defaultProps = {
+  disabled: false,
+  fullWidth: false,
+  labelPosition: 'after',
+  labelStyle: {},
+  n_clicks: 0,
+  onKeyboardFocus: () => {},
+  primary: false,
+  secondary: false,
+  setProps: () => {},
+};
+
+export default class SDFlatButton extends Component<Props> {
+  constructor(props: Props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.props.setProps({n_clicks: this.props.n_clicks + 1});
+    this.props.onClick({event: 'click'});
```

Comment:
`onClick` can be undefined here (and do we need it?), and this will break if it is clicked on without `onClick` being defined.

[Fire event](https://github.com/plotly/dash-html-components/blob/master/src/components/Button.react.js#L155) is different than `onClick` here.

Reply:
I'll change this to `fireEvent`.

Reply:
These function calls need the checks that the [function is defined](https://github.com/plotly/dash-html-components/blob/master/src/components/Button.react.js#L5)